packages/scenes-react/src/contexts/SceneContextProvider.tsx

```diff
@@ -52,7 +52,7 @@ export function SceneContextProvider({ children, timeRange, withQueryController
 
     if (parentContext) {
       getUrlSyncManager().handleNewObject(childContext);
-      parentContext.setState({ childContext });
+      parentContext.setState({ contextChildren: [...(parentContext.state.contextChildren ?? []), childContext] });
```

Comment:
```suggestion
      parentContext.addChildContext(childContext)
```

would be nice, just for readability to move that to a function on the context class

Reply:
Agreed!